vulnerabilities/weak-sources-randomness.md

```diff
@@ -1,12 +1,62 @@
 ## Weak Sources of Randomness from Chain Attributes
 
-Using chain attributes for randomness, e.g.: `block.timestamp`, `blockhash`, and `block.difficulty` can seem like a good idea since they often produce pseudo-random values. The problem however, is that Ethereum is entirely deterministic and all available on-chain data is public. Chain attributes can either be predicted or manipulated, and should thus never be used for random number generation.
+Using chain attributes for randomness, e.g.: `block.timestamp`, `blockhash`, and `block.difficulty` can seem like a good idea since they often produce pseudo-random values. The problem, however, is that Ethereum is entirely deterministic and all available on-chain data is public. Chain attributes can either be predicted or manipulated, and should thus never be used for random number generation.
 
-A common solution is to use an oracle solution such as [Chainlink VRF](https://docs.chain.link/vrf/v2/introduction/).
+### Example of Weak Randomness
+
+```solidity
+pragma solidity ^0.8.24;
+
+contract GuessTheRandomNumber {
+    constructor() payable {}
+
+    function guess(uint256 _guess) public {
+        uint256 answer = uint256(
+            keccak256(
+                abi.encodePacked(blockhash(block.number - 1), block.timestamp)
+            )
+        );
+
+        if (_guess == answer) {
+            (bool sent,) = msg.sender.call{value: 1 ether}("");
+            require(sent, "Failed to send Ether");
+        }
+    }
+}
+```
+In the above example, the answer variable is initialized using block.blockhash and now (current timestamp). This method is insecure because both block.blockhash and now can be influenced or predicted by miners, making it possible to guess the answer and win the challenge unfairly.
```

Comment:
Should wrap `block.blockhash` and `now` in backticks (``)

Reply:
Sorry for late response. I have made the changes that you suggested.